### PR TITLE
Fix BigInteger out of range bug, cause of using long value.

### PR DESCRIPTION
**Pull request by Sunshow**
Previous version use number.longValue, and may be out of range, use String to fix it just like BigDecimal.

### CODE REVIEW

**Comment by Sunshow** [comment]
@simonbasle just commit a new test case named `monoBigIntegerRange()`

**Comment by simonbasle** [comment]
for reference: this is similar to #260 and #261

File: reactor-extra/src/test/java/reactor/math/ReactorMathTests.java [hunk 1]
```diff
@@ -177,6 +177,22 @@ public void emptySumBigInteger() {
 		verifyEmptyResult(MathFlux.sumBigInteger(Mono.empty()));
 	}
 
+	@Test
+	public void monoBigIntegerRange() {
```
Comment (simonbasle) [inline reactor-extra/src/test/java/reactor/math/ReactorMathTests.java:181]: rename to `noOverflowSumBigInteger` and split out the `averageBigInteger` case into `noOverflowAverageBigInteger()` (to be placed under the `emptyAverageBigInteger` one)

**Review by simonbasle — CHANGES_REQUESTED** [review]
looking good. one last comment around tests and this is good to merge from my perspective @Sunshow 👍

**Comment by simonbasle** [comment]
if that's all good from your perspective @Sunshow I'll rebase on top of 3.4.x and merge in that branch (will need to do a force-push of your branch, FYI)

**Comment by Sunshow** [comment]
> if that's all good from your perspective @Sunshow I'll rebase on top of 3.4.x and merge in that branch (will need to do a force-push of your branch, FYI)

That's ok, looking forward to the new release.

**Comment by simonbasle** [comment]
I cannot rework the PR, either because you haven't authorized maintainers to modify your fork/PR or because this was opened from your fork's `main` branch. you will need to close that one and reopen a PR that applies the changes in a branch derived from `3.4.x` unfortunately, if you want this to ship in next week's 3.4.8 release 😞

**Comment by Sunshow** [comment]
Complete the PR.

**Comment by Sunshow** [comment]
> > Complete the PR.
> 
> I'm assuming you're asking me to add the fix in 3.4.x branch myself? I can do that and I think it makes sense.
> 
> Yet, let me know if you still want to contribute it.
> 
> In order to avoid the same confusion in the future, I would suggest always contributing PRs from feature branches (not a direct equivalent of the target maintenance branch) and making sure the "allow maintainers of the project to contribute to that PR" checkbox is checked.

My fault, that's my first PR, and I'm busy taking care of my babies, later I will learn how to continue.

**Comment by simonbasle** [comment]
that's no problem @Sunshow ! thanks for bringing the issue to our attention, in any case 🙇 👍  I will take care of that one, don't worry

**Pull request closed without merging**